### DISCUSSION TITLE
BelongsToMany::using() method not working well

### DISCUSSION THREAD

**Discussion by Ashk2a** [laravel/framework#39611] [Q&A]
- Laravel Version: 8.70.2
- PHP Version: 8.0
- Database Driver & Version: MariaDB latest & MySQL 8

### Description:

BelongsToMany::using() method not totally works. Work fine on create but not when I want get the collection through the relation.

I've two connections named **website** (default connection) and **auth**.

### Steps To Reproduce:

Models :
- [User.php (website)](https://github.com/Ashk2a/wowlf/blob/bug/belongs-to-many-using/modules/Auth/Models/User.php#L93)
- [Account.php (auth)](https://github.com/Ashk2a/wowlf/blob/bug/belongs-to-many-using/modules/Auth/Models/Account.php#L44)

Pivot :
- [UserAccount (website)](https://github.com/Ashk2a/wowlf/blob/bug/belongs-to-many-using/modules/Auth/Models/UserAccount.php#L17)

Service using the relation User->accounts :
- [AuthService](https://github.com/Ashk2a/wowlf/blob/bug/belongs-to-many-using/modules/Auth/Services/AuthService.php#L53)

So when i want create a new record everything work ! But when i want to get the collection `$user->accounts` the using looks ignored..

I've this error : (trying to use the model Account connection and not the pivot connection)
`SQLSTATE[42S02]: Base table or view not found: 1146 Table 'auth.user_accounts' doesn't exist`

Also I'm forced to precise the table in the belongsToMany cause the attribute $table in Pivot is ignored..

```php
    /**
     * @return BelongsToMany
     */
    public function accounts(): BelongsToMany
    {
        return $this->belongsToMany(Account::class, 'user_accounts')
            ->using(UserAccount::class)
            ->withTimestamps();
    }
```

**Comment by seanock** [comment]
Hey, I have also encountered this problem. Have you found a solution?
